src/HttpParser.jl

```diff
@@ -87,6 +87,59 @@ type ParserSettings
     on_chunk_complete::Ptr{Void}
 end
 
+function ParserSettings(on_message_begin_cb::Ptr{Void},
+  on_url_cb::Ptr{Void},
+  on_status_complete_cb::Ptr{Void},
+  on_header_field_cb::Ptr{Void},
+  on_header_value_cb::Ptr{Void},
+  on_headers_complete_cb::Ptr{Void},
+  on_body_cb::Ptr{Void},
+  on_message_complete_cb::Ptr{Void})
+    Base.depwarn("ParserSettings(on_url_cb, on_status_complete_cb, "
+            * " on_header_field_cb, on_header_value_cb, on_header_value_cb,"
+            * " on_headers_complete_cb, on_body_cb, on_message_complete_cb)"
+            * " is deprecated use ParserSettings(on_url_cb, on_status_complete_cb, "
+            * " on_header_field_cb, on_header_value_cb, on_header_value_cb,"
+            * " on_headers_complete_cb, on_body_cb, on_message_complete_cb,"
+            * " on_chunk_header, on_chunk_complete)", :ParserSettings)
+    function dummy(parser)
+      0
+    end
+    dummy_cb = cfunction(dummy, HTTP_CB...)
+    ParserSettings(on_message_begin_cb, on_url_cb, on_status_complete_cb,
+      on_header_field_cb, on_header_value_cb, on_headers_complete_cb,
+      on_body_cb, on_message_complete_cb, dummy_cb, dummy_cb)
+end
+
+function ParserSettings(;
+  on_message_begin_cb::Ptr{Void} = C_NULL,
+  on_url_cb::Ptr{Void} = C_NULL,
+  on_status_complete_cb::Ptr{Void} = C_NULL,
+  on_header_field_cb::Ptr{Void} = C_NULL,
+  on_header_value_cb::Ptr{Void} = C_NULL,
+  on_headers_complete_cb::Ptr{Void} = C_NULL,
+  on_body_cb::Ptr{Void} = C_NULL,
+  on_message_complete_cb::Ptr{Void} = C_NULL,
+  on_chunk_header::Ptr{Void} = C_NULL,
+  on_chunk_complete::Ptr{Void} = C_NULL)
+    function dummy(x...)
+      0
+    end
+    dummy_cb = cfunction(dummy, HTTP_CB...)
+    dummy_data_cb = cfunction(dummy, HTTP_DATA_CB...)
```

Comment:
According to CI this works fine on 0.4. Any reason it shouldn't work? You mean I should move the function `dummy` outside of constructor? That would work, but will pollute the HttpParser namespace and I assume that LLDB is smart enough to make this fast anyway.